pkg/agent/agentclient/client.go

```diff
@@ -47,15 +60,14 @@ func (c *connContext) cleanup() {
 	c.cleanOnce.Do(c.cleanFunc)
 }
 
-func NewAgentClient(address string) *AgentClient {
-	a := &AgentClient{
-		connContext: make(map[int64]*connContext),
-		address:     address,
-	}
-
-	return a
-}
-
+// Connect connnects to proxy server to establish a gRPC stream,
```

Comment:
We need to think about abstracting away the gRPC piece of this. We already know we will need to support other types of streams in the near future.

Reply:
right. we don't have a common interface for client. this method only works for gRPC.

Reply:
Agreed. However it is not actually part of the gRPC client code. So I believe it will eventually have to support non gRPC streams as well.